### ISSUE TITLE
🐛 BUG: <Markdown> component processes inline markup incorrectly

### ISSUE THREAD

**Issue by telic** [snowpackjs/astro#493]
### Bug Report: Quick Checklist

- [x] I am on the latest version of Astro (0.13.10) & all plugins
- [x] I am on an ESM Node.js version 16.2.0
- [x] I use package manager: npm
- [x] I run Astro on MacOS

### Describe the bug

The `<Markdown>` component renders the content of inline-level HTML and components as if they were block-level.

### Steps to reproduce

Render the following .astro component:

```astro
---
import { Markdown } from 'astro/components';
---
<Markdown>
  Inline *markdown **constructs** nest* nicely, and remain inline.

  Inline <em>HTML markup **does not** stay inline</em>.
</Markdown>
```

### Expected versus actual behavior

The expected rendered result fragment should be:

```html
<p>Inline <em>markdown <strong>constructs</strong> nest</em> nicely, and remain inline.</p>
<p>Inline <em>HTML markup <strong>does not</strong> stay inline</em>.</p>
```

However, the actual rendered output (with classes removed and whitespace simplified) is:
```html
<p>Inline <em>markdown <strong>constructs</strong> nest</em> nicely, and remain inline.</p>
<p>Inline</p><em><p>HTML markup <strong>does not</strong> stay inline</p></em><p>.</p>
```

That is, the inline element is rendered in a way that ignores its inline context, and ends up introducing superfluous paragraph elements.

The same bug applies when the inlined element is a custom component.

**Comment by jasikpark** [comment]
Here's an example of how it should be parsed: https://spec.commonmark.org/dingus/?text=%20%20Inline%20*markdown%20**constructs**%20nest*%20nicely%2C%20and%20remain%20inline.%0A%0A%20%20Inline%20%3Cem%3EHTML%20markup%20**does%20not**%20stay%20inline%3C%2Fem%3E.

Testing with the CommonMark online playground!

**Comment by natemoo-re** [comment]
`astro@next` fixes a lot of Markdown bugs by leaning on [MDX](https://mdxjs.com) as much as possible. 

Here's the behavior on `astro@0.21.0-next.2`:
https://stackblitz.com/edit/github-blyzf6?file=src%2Fpages%2Findex.astro

In this case, it looks like there's a smaller issue where the second paragraph is processed as plaintext and doesn't have the wrapping `<p></p>` tags.

**Comment by jasikpark** [comment]
Could you post a screenrecording of the stackblitz? It doesn't work in Firefox 😅

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
